### PR TITLE
List app status

### PR DESCRIPTION
**Pull request by diegoribero**
Adds a status column to `matrix list apps`, this will show the current state of each app.

**active**: Running
**inactive**: Installed, not running
**pending**: Currently installing
**error**: Failed installation

### CODE REVIEW

**Comment by eighteyes** [comment]
Please describe what this is for.

File: bin/matrix-list.js [hunk 1]
```diff
@@ -30,12 +30,20 @@ Matrix.localization.init(Matrix.localesFolder, Matrix.config.locale, function ()
...
+      Matrix.firebase.app.list(function (err, apps) {
         Matrix.loader.stop();
         if (err) return console.error('- ', t('matrix.list.app_list_error') + ':', err);
-        if (_.isUndefined(data)) data = {};
-        console.log(Matrix.helpers.displayApps(data));
-        process.exit();
+        if (_.isUndefined(apps)) apps = {};
+        async.forEach(Object.keys(apps), function (appId, done) { 
+          Matrix.firebase.app.getStatus(appId, function (status) { 
+            console.log("STATUS WATCH TRIGGERED! " + Matrix.config.user.id + '>' + Matrix.config.device.identifier + '>' + appId + '>' + status);
```
Comment (diegoribero) [inline bin/matrix-list.js]: Description updated, added inline comments, cleaned a bit, not sure how that console.log slipped through 😁

**Comment by brianofrokk3r** [comment]
@degoribero

**Comment by brianofrokk3r** [comment]
#34

File: bin/matrix-list.js [hunk 2]
```diff
@@ -30,12 +30,24 @@ Matrix.localization.init(Matrix.localesFolder, Matrix.config.locale, function ()
...
+      Matrix.firebase.app.list(function (err, apps) {
         Matrix.loader.stop();
         if (err) return console.error('- ', t('matrix.list.app_list_error') + ':', err);
-        if (_.isUndefined(data)) data = {};
-        console.log(Matrix.helpers.displayApps(data));
-        process.exit();
+        if (_.isUndefined(apps)) apps = {};
+
+        //Retrieve status for each app
+        async.forEach(Object.keys(apps), function (appId, done) {
```
Comment (eighteyes) [inline bin/matrix-list.js]: I don't understand. Shouldn't `meta.status` be available on `apps`? Why are we doing all these lookups?

Comment (eighteyes) [inline bin/matrix-list.js]: Oh nm. `list` returns from `user`.  got it

**Pull request merged**